pkg/ruler/frontend_decoder.go

```diff
@@ -86,3 +97,51 @@ func (j JsonDecoder) scalarToPromQLVector(scalar model.Scalar) promql.Vector {
 		Metric: labels.Labels{},
 	}}
 }
+
+func (p ProtobufDecoder) ContentType() string {
+	return "application/x-protobuf"
+}
+
+func (p ProtobufDecoder) Decode(body []byte) (promql.Vector, Warning, error) {
+	resp := tripperware.PrometheusResponse{}
+	if err := resp.Unmarshal(body); err != nil {
+		return nil, nil, err
+	}
+
+	switch data := resp.Data.Result.Result.(type) {
```

Comment:
Can we check `resp.Data.ResultType` instead? This way we can probably get rid of scalar

Reply:
Ok, that would work. Thanks!